app/models/miq_server.rb

```diff
@@ -53,6 +55,10 @@ class MiqServer < ApplicationRecord
 
   RESTART_EXIT_STATUS = 123
 
+  def validate_zone_visible?
```

Comment:
This feels like it should be `validate_non_maintenance_zone?` or something?  This feels like we're making `!visible == maintenance` which isn't necessarily true.

Reply:
maintenance is subset of visible. We can create and use many invisible zones for various purposes and functions. These zones won't be visible by UI. And invisible zones cannot be assigned to servers

Reply:
I agree on this last part.  The only thing we really know right now and is what the whole BZ wants to solve is that no servers are in the maintenance zone, so we can temporarily move our providers to that zone and have no concern with any of our processes trying to access the provider.

Again, I'm not sure where visibility comes into this.  How the UI presents maintenance zone objects is an entirely different thing and shouldn't be addressed here.

Reply:
@agrare ok, I have no problem with it, okay
@jrafanie I hope I explained difference in previous comment

Reply:
@slemrmartin Sorry, I still have no idea where visibility came from or what it means.  Is there another BZ or PR that explains it?  The only requirement I see is maintenance mode.  I guess that's fine but I can't really help review something if I don't understand it.